### PR TITLE
I added the setBoard route

### PR DESCRIPTION
**Pull request by dgibb**
allows you to set the board to any state as long as a game is started from either client using a post request

example code to set board from the console

`const newBoard = {
   Board:[
     [0, 0, 0, 0, 0, 0, 0, 0],
     [0, 0, 0, 0, 0, 0, 0, 0],
     [0, 0, 0, 0, 0, 0, 0, 0],
     [0, 0, 0, 0, 0, 0, 0, 0],
     [0, 0, 0, 0, 0, 0, 0, 0],
     [0, 0, 0, 0, 0, 0, 0, 0],
     [0, 0, 0, 0, 0, 0, 0, 0],
     [0, 0, 0, 0, 0, 0, 0, 0],
    ]
}`

`const jsonString = JSON.stringify(newBoard)`

`$.post('/setboard', jsonString);`

### CODE REVIEW

File: Handler.go [hunk 1]
```diff
@@ -129,6 +132,32 @@ func events(w http.ResponseWriter, r *http.Request) {
...
+	if err != nil {
+		panic(err)
+	}
+	theGame.board = newBoard.Board
+	move.Turn = theGame.turn
+	hasMove, board := findPotentialMoves(theGame.board, move.Turn)
+	if hasMove {
+		move.Board = board
+	} else {
+		//seems like a sloppy fix
```
Comment (Rogibb111) [inline Handler.go:151]: I agree with this comment....what are you trying to do here.

File: Handler.go [hunk 2]
```diff
@@ -16,10 +16,14 @@ type moveResponse struct {
 }
 
 type newGameResponse struct {
-	Full   bool
+	Full bool
 	Player int
 }
 
+type newBoard struct {
```
Comment (Rogibb111) [inline Handler.go:23]: Is this struct necessary? I cant think off the top of my head how you would do it differently, but I feel like there might be an even easier way to reset the board.

**Review by Rogibb111 — APPROVED** [review]

File: Handler.go (hunk 1 above)
Comment (dgibb) [reply Handler.go:151]: It's supposed to replace the board regardless of the value of hasMove, I was thrown an error when hasMove was assigned a value but never used. I may have misinterpreted the original code though, I assumed that hasMove was supposed to make it remain the original players turn if after they made a move the other player had no valid moves.

File: Handler.go (hunk 2 above)
Comment (dgibb) [reply Handler.go:23]: potentially not sending the board as json and using raw data instead. The handler from my gba server is just

>buf, err := ioutil.ReadAll(r.Body)
>if err!=nil {log.Fatal("request",err)}
>ROM = buf

sends and receives a 1D byte array, that may of may not work for a 2D int array

**Pull request merged**